### ISSUE TITLE
Very slow test filtering

### ISSUE THREAD

**Issue by hurricane642** [perturbo-code/perturbopy#12]
PROBLEM:
Filtering tests (by names, tags, or ephr file names) takes a considerable amount of time. A full perturbo-qe2pert-perturbo testing takes 15 minutes, and filtering can take 2 minutes. Not critical, but annoying. Also, can be critical in future, when the number of tests will be increased;
SUGGESTION:
Rewrite the `filter_tests` function from `perturbopy.test_utils.run_test.run_utils` . It may be necessary to avoid a lot of for loops;

EDIT:
A closer inspection revealed that when passing the `test_names` parameter the filtering speed is more than high, while in the case of `epwan/tags` the filtering time is about 11-12 seconds, which, although unpleasant, is not critical. So I decrease the priority of the issue

**Comment by hurricane642** [comment]
Solved (as it was possible) in #52

**Comment by hurricane642** [comment]
Hey, after the discussion with @imaliyov we decided to work a bit more on this part of code. 
I've made a measurement of time consumption of different part of code inside of the `filter_tests` function. As a results, I've obtained, that function `read_test_tags` consume almost 90% of time (!!!!). 
Also, I've checked what `Loader` is used for the yaml file - it's an efficient `CLoader`. So, I'll try to change the format of our yaml-files in order to speed up the test filtering
